### PR TITLE
Fix: Disconnecting

### PR DESCRIPTION
**Pull request by Sunnickel**
Looked into the bug again in [this issue](https://github.com/ferrumc-rs/ferrumc/issues/212) and my older [PR](https://github.com/ferrumc-rs/ferrumc/pull/219). I noticed that the bug wasn't completely fixed. While I fixed some parts, others were completely ignored somehow. Probably this time too, either didn't see it or didn't test for enough edge-cases.

### connection_killer
refactored for better borrowing

### new_connection
added a disconnect_handle to the entity_commands

### connection
Heres where the magic happens. Added the ServerState and the Entity to the StreamWriter Struct. In new() we now handle that Entities can be removed on a writing error and after disconnecting there is a nice clean-up to ensure everything gets removed from ecs and the connection. 

Also on ReCore's wish I removed the kill() method.

## Motivation and Context
Worked on this bug before and there were still errors.

https://github.com/ferrumc-rs/ferrumc/issues/212

## How has this been tested?
Tested via cargo test and tested it with 2 minecraft instances running and connecting. 

Tested on Windows 11, RustRover, 2 Vanilla Minecraft instances

## Screenshots (if appropriate):
-

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (restructuring code, without changing its behavior)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] Clippy passes with no warnings.

### CODE REVIEW

**Comment by Sunnickel** [comment]
https://github.com/ferrumc-rs/ferrumc/pull/223/commits/ad5af47d326407b0259f4f14e8ff32b689c8f4d0 is from @Tonguechaude

File: src/bin/src/systems/connection_killer.rs [hunk 1]
```diff
@@ -12,48 +12,52 @@ pub fn connection_killer(
...
+            .map(|(_, _, identity)| identity.clone());
+
+        if disconnecting_player_identity.is_none() {
+            warn!("Player's entity has already been removed");
+            continue;
+        }
+
+        let disconnecting_player_identity = disconnecting_player_identity.unwrap();
+
+        for (entity, conn, player_identity) in query.iter() {
```
Comment (evelynmeowww) [inline src/bin/src/systems/connection_killer.rs:27]: what's the point in doing this when you're already calling get earlier and only taking the identity?

Comment (Sunnickel) [reply src/bin/src/systems/connection_killer.rs:27]: get gives me the entity that wants to disconnect current as a option tho, before the loop i just get the object if there is something in there.

down i get every player that is currently on the server, I need this to send the messages to players if someone disconnects.

**Comment by Mcrtin** [comment]
can't reproduce on fix/fix-disconnect; this pr seems to fix the issue

**Comment by ReCore-sys** [comment]
Cool, in that case I'll merge
